gitops/kcp/gateway/haproxy-deployment.yaml

```diff
@@ -31,6 +34,30 @@ spec:
               mountPath: /usr/local/etc/haproxy/haproxy.cfg
               subPath: haproxy.cfg
               readOnly: true
+          livenessProbe:
+            failureThreshold: 3
+            successThreshold: 1
+            initialDelaySeconds: 0
+            timeoutSeconds: 1
+            tcpSocket:
+              port: 80
+            periodSeconds: 10
```

Comment:
When the router is under load you will kill it. This will make things worse.
Understanding what is an error state vs the process not being responsive because of its business is not a trivial thing.
If you are interested you can check what has been done with the OpenShift router to try to get something meaningful:
https://github.com/openshift/router/blob/master/pkg/router/metrics/health.go